src/transformers/models/swin/modeling_swin.py

```diff
@@ -416,17 +448,23 @@ def __init__(self, config, dim, input_resolution, num_heads, shift_size=0):
             attn_mask = attn_mask.masked_fill(attn_mask != 0, float(-100.0)).masked_fill(attn_mask == 0, float(0.0))
         else:
             attn_mask = None
+        return attn_mask
 
-        self.attn_mask = attn_mask
-
-    def forward(self, hidden_states, head_mask=None, output_attentions=False):
-        height, width = self.input_resolution
+    def forward(self, hidden_states, input_dimensions, head_mask=None, output_attentions=False):
+        height, width = input_dimensions
         batch_size, dim, channels = hidden_states.size()
         shortcut = hidden_states
 
         hidden_states = self.layernorm_before(hidden_states)
         hidden_states = hidden_states.view(batch_size, height, width, channels)
-
+        # pad feature maps to multiples of window size
+        pad_l = pad_t = 0
+        pad_r = (self.window_size - width % self.window_size) % self.window_size
+        pad_b = (self.window_size - height % self.window_size) % self.window_size
+        pad_value = (0, 0, pad_l, pad_r, pad_t, pad_b)
```

Comment:
if l, r, b and t represent left, rgiht, bottom and top, they should be changed to that. The code would be more readable with those.

Reply:
fixed, all pad related code was also moved to methods called `maybe_pad`